fvm/src/executor/default.rs

```diff
@@ -123,17 +124,52 @@ where
             });
 
             let result = cm.with_transaction(false, |cm| {
+                let pl = cm.context().price_list;
+
+                if let Some(sender_state) = sender_st {
+                    if cm
+                        .machine()
+                        .builtin_actors()
+                        .is_eoa_actor(&sender_state.code)
+                    {
+                        // We need to deploy the Ethereum EOA actor
+                        cm.charge_gas(pl.on_create_actor(false))?;
+
+                        // Transform the actor code CID into the Ethereum EOA code CID.
+                        let eoa_code_cid = *cm.machine().builtin_actors().get_eoa_code();
+                        cm.state_tree_mut().set_actor(
+                            sender_id,
+                            ActorState {
+                                code: eoa_code_cid,
+                                ..sender_state
+                            },
+                        )?;
+
+                        // Now actually invoke the constructor
+                        let ret = cm.send::<K>(
+                            SYSTEM_ACTOR_ID,
+                            msg.from,
+                            METHOD_CONSTRUCTOR,
+                            None,
+                            &TokenAmount::zero(),
+                            None,
+                        )?;
+
+                        // If we _failed_ to deploy the EOA actor, we abort early.
+                        // If we succeeded, proceed to the actual invocation.
+                        if !ret.exit_code.is_success() {
+                            return Ok(ret);
+                        }
```

Comment:
Should exit with `SYS_SENDER_INVALID` (or a new exit code). We can't return the exit code from this actor, because, if this fails, this actor doesn't actually exist at this address and this call never happened...

Reply:
Yeah, you're right.